apps/src/codegen.js

```diff
@@ -29,9 +29,11 @@ exports.evalWith = function (code, options, legacy) {
     // interpret the JS program all at once:
     myInterpreter.run();
   } else if (!legacy) {
-    new PatchedInterpreter(`(function () { ${code} })()`, (interpreter, scope) => {
+    const myInterpreter = new PatchedInterpreter(`(function () { ${code} })()`, (interpreter, scope) => {
       marshalNativeToInterpreterObject(interpreter, options, 5, scope);
-    }).run();
+    });
+    myInterpreter.run();
+    return myInterpreter;
```

Comment:
You should rebase onto staging. `evalWith` now has a proper doc block and unit tests. Make sure to update the doc block and the tests with the new return behavior. It's so confusing that the type of the return value changes _a lot_ depending on whether legacy is true or not. I guess hopefully legacy will eventually go away. We should probably just create a separate function called `evalWith_DEPRECATED()` to do the "legacy" stuff.